autotest/test_modeltime.py

```diff
@@ -0,0 +1,292 @@
+        (11, None): datetime.datetime(2024, 12, 28, 23, 59, 58),
+    }
+
+    nrec = 12
+    perlen = np.full((nrec,), 30.25)
+    nstp = np.full((nrec,), 4, dtype=int)
+    tslen = np.full((nrec,), 1.25)
+    start_datetime = "2023-12-31t23:59:59"
+    time_unit = "days"
+
+    mt = ModelTime(
+        perlen, nstp, tslen, time_units=time_unit, start_datetime=start_datetime
+    )
+
+    for (kper0, kstp0), dt in validate.items():
+        if kstp0 is None:
+            kper, _ = mt.intersect(dt)
+            if kper != kper0:
+                raise AssertionError("intersect() not returning correct stress-period")
+
+        else:
+            kper, kstp = mt.intersect(dt)
+            if kper != kper0 or kstp != kstp0:
+                raise AssertionError(
+                    "intersect() not returning correct stress-period and timestep"
+                )
+
+
+def test_totim_intersect():
+    validate = {(0, None): 30.2, (1, 3): 60.4, (4, 0): 126.23, (11, None): 362.9}
+    nrec = 12
+    perlen = np.full((nrec,), 30.25)
+    nstp = np.full((nrec,), 4, dtype=int)
+    tslen = np.full((nrec,), 1.25)
+    start_datetime = "2023-12-31t23:59:59"
+    time_unit = "days"
+
+    mt = ModelTime(
+        perlen, nstp, tslen, time_units=time_unit, start_datetime=start_datetime
+    )
+
+    for (kper0, kstp0), totim in validate.items():
+        if kstp0 is None:
+            kper, _ = mt.intersect(totim=totim)
+            if kper != kper0:
+                raise AssertionError("intersect() not returning correct stress-period")
+
+        else:
+            kper, kstp = mt.intersect(totim=totim)
+            if kper != kper0 or kstp != kstp0:
+                raise AssertionError(
+                    "intersect() not returning correct stress-period and timestep"
+                )
+
+
+def test_mf2005_modeltime():
+    nlay = 1
+    nrow = 9
+    ncol = 9
+    delc = 10
+    delr = 10
+    top = np.full((nrow, ncol), 100)
+    botm = np.zeros((nlay, nrow, ncol))
+    idomain = np.ones(botm.shape, dtype=int)
+    strt = np.full(botm.shape, np.max(top) - 5)
+    nper = 5
+    nstp = [5, 4, 5, 5, 5]
+    perlen = [31, 28, 31, 30, 31]
+    start_datetime = datetime.datetime(2024, 1, 1)
+    start_datetime_str = "1/1/2024"
+
+    ml = flopy.modflow.Modflow(modelname="dev_time", model_ws="dev_time")
+
+    dis = flopy.modflow.ModflowDis(
+        ml,
+        nlay=nlay,
+        nrow=nrow,
+        ncol=ncol,
+        nper=nper,
+        delc=delc,
+        delr=delr,
+        top=top,
+        botm=botm,
+        perlen=perlen,
+        nstp=nstp,
+        steady=False,
+        itmuni=4,
+        lenuni=2,
+        start_datetime=start_datetime_str,
+    )
+    bas = flopy.modflow.ModflowBas(ml, ibound=idomain, strt=strt)
+
+    modeltime = ml.modeltime
+    if modeltime.start_datetime != start_datetime:
+        raise AssertionError("start_datetime improperly stored")
+
+    result = modeltime.intersect("3/06/2024 23:59:59")
```

Comment:
I know `intersect` is used elsewhere, but consider renaming this method to say what it does, for example ``get_kstpkper()`` (if I'm understanding this correctly)